sdk/containerregistry/azure-containerregistry/azure/containerregistry/_download_stream.py

```diff
@@ -68,7 +68,7 @@ def __next__(self) -> bytes:
         try:
             return self._yield_data()
         except StopIteration:
-            if self._downloaded >= self._blob_size:
+            if self._downloaded == self._blob_size:
```

Comment:
what if self._downloaded > self._blob_size?

Reply:
- self._downloaded should always `<=` self.blob_size. 
- downloaded is possible larger than blob_size when downloading a chunk more than one time, but we don't support retry when downloading blob.
So I think `>` should not happen.

Reply:
Revert the change, it will keep downloading when downloaded data is larger than blob_size in some cases.